lib/model.js

```diff
@@ -2108,29 +2109,33 @@ Model.insertMany = function(arr, options, callback) {
   }
 
   var toExecute = [];
-  arr.forEach(function(doc) {
+  var validationError = new ValidationError(); // Group all ValidationErrors together
+  arr.forEach(function(doc, index) {
     toExecute.push(function(callback) {
       doc = new _this(doc);
       doc.validate({ __noPromise: true }, function(error) {
         if (error) {
-          // Option `ordered` signals that insert should be continued after reaching
-          // a failing insert. Therefore we delegate "null", meaning the validation
-          // failed. It's up to the next function to filter out all failed models
-          if (options != null && typeof options === 'object' && options['ordered'] === false) {
-            return callback(null, null);
-          }
-          return callback(error);
+          validationError.errors[index] = error;
+          // Never return the error, as it will cause parallelLimit to immediately
+          // abort. Return null doc for `ordered: false` option.
+          return callback(null, null);
         }
         callback(null, doc);
       });
     });
   });
 
   parallelLimit(toExecute, limit, function(error, docs) {
-    if (error) {
-      callback && callback(error);
+    // `error` should never be anything but null.
+
+    // Option `ordered: false` signals that insert should be continued after reaching
+    // a failing insert. Return grouped ValidationError if this option doesn't exist
+    // or is true (if there are any).
+    if ((options == null || typeof options !== 'object' || options['ordered'] === true) && Object.keys(validationError.errors).length > 0) {
```

Comment:
`ordered` is true by default: https://docs.mongodb.com/manual/reference/method/db.collection.insertMany/#db.collection.insertMany so we shouldn't return multiple validation errors if `ordered` isn't set